File: Java/SpectableJavaExamplesTest/src/main/java/production/Match.java
package production;

import java.util.Objects;

public class Match {
    public final String matchedAddress;
    public final AddressComponents addressComponents;

    public Match(String matchedAddress, AddressComponents addressComponents) {
        this.matchedAddress = matchedAddress;
        this.addressComponents = addressComponents;
    }

    public static class Builder {
        private String matchedAddress;
        private AddressComponents addressComponents;

        public Builder matchedAddress(String matchedAddress) { this.matchedAddress = matchedAddress; return this; }
        public Builder addressComponents(AddressComponents addressComponents) { this.addressComponents = addressComponents; return this; }

        public Match build() {
            return new Match(matchedAddress, addressComponents);
        }
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Match)) return false;
        Match that = (Match) o;
        return Objects.equals(matchedAddress, that.matchedAddress)
            && Objects.equals(addressComponents, that.addressComponents);
    }

    @Override
    public int hashCode() {
        return Objects.hash(matchedAddress, addressComponents);
    }

    @Override
    public String toString() {
        return "Match{" + "matchedAddress=" + matchedAddress + ", " + "addressComponents=" + addressComponents + "}";
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/production/Match.cs
namespace production
{
    public class Match
    {
        public string MatchedAddress { get; }
        public AddressComponents AddressComponents { get; }

        public Match(string matchedAddress, AddressComponents addressComponents)
        {
            MatchedAddress = matchedAddress;
            AddressComponents = addressComponents;
        }

        public class Builder
        {
            private string _matchedAddress;
            private AddressComponents _addressComponents;

            public Builder MatchedAddress(string value) { _matchedAddress = value; return this; }
            public Builder AddressComponents(AddressComponents value) { _addressComponents = value; return this; }

            public Match Build() =>
                new Match(_matchedAddress, _addressComponents);
        }
    }
}
